fuse/context.py

```diff
@@ -183,10 +212,13 @@ def set_simulation_config_file(context, config_file_name):
         for option_key, option in plugin.takes_config.items():
             if isinstance(option.default, str) and "SIMULATION_CONFIG_FILE.json" in option.default:
                 context.config[option_key] = option.default.replace(
-                    "SIMULATION_CONFIG_FILE.json",
-                    config_file_name,
+                    "SIMULATION_CONFIG_FILE.json", config_file_name
                 )
 
+            # Special case for the photoionization_modifier
```

Comment:
Why `photoionization_modifier` is special?

Reply:
Hi @dachengx, this part is a little trick to properly set the `photoionization_modifier` config argument. The problem is that it is not written to the context config by `st.apply_xedocs_configs(version=corrections_version)` ([here](https://github.com/XENONnT/fuse/blob/0f81e47988275f866efb223bd28f0756dc7ea675/fuse/context.py#L178-L179)). As as result the `plugin.run_id` would not be overwritten by the `corrections_run_id` [here](https://github.com/XENONnT/fuse/blob/0f81e47988275f866efb223bd28f0756dc7ea675/fuse/context.py#L183-L190). 

I solved this problem by explicitly writing this config argument to the context config in the `set_simulation_config_file` function as we loop over all plugins and plugin configs anyways so we will also encounter the `photoionization_modifier` there. As it is now in the context config, `plugin.run_id` can now be changed to `corrections_run_id`.

Reply:
Will this be solved if the `photoionization_modifier` is added to the global config: https://github.com/XENONnT/corrections/blob/92aa12f5fbae460c990e04bc571af0df3f67e68d/XENONnT/global_versions/global_v14.json?

Reply:
I think so yes. Then `st.apply_xedocs_configs(version=corrections_version)` should already take care of this.